### PR TITLE
chore: use go generate to generate test PKIs

### PR DESCRIPTION
**Pull request by Baha-sk**
Currently to test the project, users are forced to call `make generate-test-keys`
    
This target is not intended to be used by users of the library.
    
Instead a more general `go generate` command can be called to create test keys

closes issue #13

### CODE REVIEW

File: genTestPKI.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
```
Comment (troyronda) [inline genTestPKI.go]: Also, if these are fixtures for the http transport testing then this go:generate should be in the transport/http package.

File: scripts/openssl_env.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Copyright SecureKey Technologies Inc. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+OPENSSL_CMD=${OPENSSL_CMD:-openssl}
```
Comment (llorllale) [inline scripts/openssl_env.sh]: @Baha-sk does this command need to be configurable?

**Comment by llorllale** [comment]
@Baha-sk @troyronda if this PR is merged as-is, and #9 is also merged, it will leave the `make generate-test-keys` target

File: scripts/openssl_env.sh (hunk 2 above)
Comment (troyronda) [reply scripts/openssl_env.sh]: I generally think it’s a good pattern to allow overriding of commands in these scripts.

File: genTestPKI.go (hunk 1 above)
Comment (Baha-sk) [reply genTestPKI.go]: this means one will have to run `go generate` in transport/http package.

with the current change, we can run `go generate` from the root of the project and the artifacts will be generated in /test folder

**Comment by Baha-sk** [comment]
> @Baha-sk @troyronda if this PR is merged as-is, and #9 is also merged, it will leave the `make generate-test-keys` target

yes this is why this PR should be merged after #9 is merged, I created branch off of #9, I should have created it off of master to remove the dependency

File: genTestPKI.go (hunk 1 above)
Comment (troyronda) [reply genTestPKI.go]: The generate goes with the code that needs it, not at the base of the project.

File: scripts/openssl_env.sh (hunk 2 above)
Comment (llorllale) [reply scripts/openssl_env.sh]: I think it's a preemptive solution to a rare, user-specific problem, with the side effect of adding clutter to our scripts. Our build certainly doesn't require it - we have more control over this by using `frapsoft/openssl`.

That said, it's just a one-liner, although it leaves one (uselessly) wondering why it's configurable.

File: genTestPKI.go (hunk 1 above)
Comment (Baha-sk) [reply genTestPKI.go]: file removed, generate is now under the http transport package

File: README.md [hunk 3]
```diff
@@ -10,3 +10,16 @@
 ## License
 
 Hyperledger Aries Framework Go is licensed under the [Apache License Version 2.0](LICENSE).
+
```
Comment (rolsonquadras) [inline README.md]: This should go under proper sub-heading. Currently, it's inside Licence.

File: scripts/check_unit.sh [hunk 4]
```diff
@@ -12,4 +12,11 @@ echo "Running $0"
 PKGS=`go list github.com/hyperledger/aries-framework-go/... 2> /dev/null | \
                                                  grep -v /mocks`
 
+CWD=$PWD
+if [[ ! -d "pkg/didcomm/transport/http/testdata" ]]; then
```
Comment (fqutishat) [inline scripts/check_unit.sh]: do we need to add every package that use go generate here ?

Comment (fqutishat) [reply scripts/check_unit.sh]: use go generate ./...

File: README.md (hunk 3 above)
Comment (Baha-sk) [reply README.md]: added sub-heading

File: scripts/check_unit.sh (hunk 4 above)
Comment (Baha-sk) [reply scripts/check_unit.sh]: done

File: scripts/generate_test_keys.sh [hunk 5]
```diff
@@ -8,43 +8,43 @@
...
 #extract pubkey 2
-openssl x509 -inform pem -in test/fixtures/keys/ec-pubCert2.pem -pubkey -noout > test/fixtures/keys/ec-pubKey2.pem
+openssl x509 -inform pem -in testdata/ec-pubCert2.pem -pubkey -noout > testdata/ec-pubKey2.pem
 
 #create creds 3
-openssl ecparam -name prime256v1 -genkey -noout -out test/fixtures/keys/ec-key3.pem
-openssl req -new -key test/fixtures/keys/ec-key3.pem -subj "/C=CA/ST=ON/O=Example Inc.:aries-framework-go/OU=aries-framework-go/CN=*.example.com" -reqexts SAN -config ./test/fixtures/openssl/openssl.cnf -out test/fixtures/keys/ec-key3.csr
-openssl x509 -req -in test/fixtures/keys/ec-key3.csr -extensions SAN -extfile ./test/fixtures/openssl/openssl.cnf -CA test/fixtures/keys/ec-cacert.pem -CAkey test/fixtures/keys/ec-cakey.pem -CAcreateserial -out test/fixtures/keys/ec-pubCert3.pem -days 365
+openssl ecparam -name prime256v1 -genkey -noout -out testdata/ec-key3.pem
+openssl req -new -key testdata/ec-key3.pem -subj "/C=CA/ST=ON/O=Example Inc.:aries-framework-go/OU=aries-framework-go/CN=*.example.com" -reqexts SAN -config ../../../../test/fixtures/openssl/openssl.cnf -out testdata/ec-key3.csr
```
Comment (fqutishat) [inline scripts/generate_test_keys.sh]: why openssl.cnf in test/fixtures/openssl can we create dynamically

Comment (troyronda) [reply scripts/generate_test_keys.sh]: (Or put it in testdata folder)

File: .gitignore [hunk 6]
```diff
@@ -17,7 +17,8 @@ vendor
 
 # Test binary, build with `go test -c`
 *.test
-test/fixtures/keys
+# Generate test PKIs with `go generate`
+pkg/didcomm/transport/http/testdata
```
Comment (troyronda) [inline .gitignore]: I would put the generated certs in a subdirectory of testdata and only ignore that directory.

Comment (Baha-sk) [reply .gitignore]: this means testdata will be committed to git...

Comment (Baha-sk) [reply .gitignore]: done, new subdirectory called crypto/ is added

File: scripts/generate_test_keys.sh (hunk 5 above)
Comment (Baha-sk) [reply scripts/generate_test_keys.sh]: deleted openssl.cnf file .. it is now temporarily generated in the script (and deleted at the end)

File: pkg/didcomm/transport/http/outbound.go [hunk 7]
```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+//go:generate ../../../../scripts/openssl_env.sh ../../../../scripts/generate_test_keys.sh
```
Comment (troyronda) [inline pkg/didcomm/transport/http/outbound.go]: Why not put this under the testdata directory too?

Comment (Baha-sk) [reply pkg/didcomm/transport/http/outbound.go]: sure, might as well since they're used for http transport..

**Pull request merged**